src/main/kotlin/com/exactpro/th2/sim/template/rule/KotlinFIXRule.kt

```diff
@@ -37,20 +39,20 @@ class KotlinFIXRule(field: Map<String, Any?>) : MessageCompareRule() {
     private val aliasdc2 = "dc-demo-server2"
 
     companion object {
-        private var orderId = AtomicInteger(0)
-        private var execId = AtomicInteger(0)
-        private var TrdMatchId = AtomicInteger(0)
+        private val orderId = AtomicInteger(0)
+        private val execId = AtomicInteger(0)
+        private val TrdMatchId = AtomicInteger(0)
 
-        private var incomeMsgList = arrayListOf<ParsedMessage>()
-        private var ordIdList = arrayListOf<Int>()
+        private val buyOrdersAndIds: Queue<Pair<ParsedMessage, Int>> = LinkedList()
+        private val sellOrders: Queue<Pair<ParsedMessage, Int>> = LinkedList()
```

Comment:
I believe those collections should be thread-safe. There is not guarantee that the rule always will be called from one thread

Reply:
This is not enough to make whole rule thread-safe. We can make `handle` an `reset` methods synchronized instead.

Reply:
I think this means we did something wrong with the rule) Why do we need to have all methods synchronized?

Reply:
Okay, that means that only this block should be executed under a lock along with writes to those queues